hydrator-common/src/main/java/io/cdap/plugin/common/db/AbstractDBConnector.java

```diff
@@ -253,15 +270,51 @@ protected BrowseDetail listTables(Connection connection, @Nullable String schema
     return browseDetailBuilder.setTotalCount(count).build();
   }
 
-  protected Connection getConnection() {
+  private void validateDatabase(String database, Connection connection) throws SQLException {
+    if (database == null) {
+      return;
+    }
+    try (ResultSet catalogs = queryDatabases(connection)) {
+      boolean exits = false;
+      while (catalogs.next()) {
+        if (database.equals(catalogs.getString(RESULTSET_COLUMN_TABLE_CAT))) {
+          exits = true;
+          break;
+        }
+      }
+      if (!exits) {
+        throw new IllegalArgumentException(String.format("Database %s does not exist.", database));
+      }
+    }
+  }
+
+  protected void validateSchema(String database, String schema, Connection connection)
+    throws SQLException {
+    if (schema == null) {
+      return;
+    }
+    // NOTE Oracle schema name is case sensitive here
+    try (ResultSet schemas = connection.getMetaData()
+      .getSchemas(database == null ? connection.getCatalog() : database, schema)) {
+      if (!schemas.next()) {
+        throw new IllegalArgumentException(String.format("Schema '%s' does not exist.", schema));
+      }
+    }
+  }
+
+  protected Connection getConnection(@Nullable DBConnectorPath path) {
```

Comment:
better to not allow this to be nullable, just have a method getConnection() which returns the config connection string. Also the path is not used, is the specific plugin supposed to create a connection string from the path?

Reply:
yes, I make it Nullable on purpose so that I can have same code for both generic and specific DB plugins. 
Generic DB plugin don't need path to create a connection while specific DB plugins need

Reply:
added a empty parameter getConnection()